utils/providerAPI/getUserRepositories.ts

```diff
@@ -64,37 +59,81 @@ type BitbucketRepoObj = {
 
 // type GitlabRepoObj = {
 // }
-
-const getUserRepositoriesForGitHub = async (access_key: string, authId?: string) => {
+const getUserRepositoriesForGitHub = async (access_token: string, authId?: string) => {
 	const perPage = 100;
-	let pageNo = 1;
-	const allGitHubRepositories: RepoIdentifier[] = []
-	let isResponseFull = true;
+	let endCursor = null;
+	let hasNextPage = true;
+	const allGitHubRepositories: RepoIdentifier[] = [];
 
 	do {
-		const endPoint = `/user/repos?sort=updated&per_page=${perPage}&page=${pageNo}`;
-		const { data: repos }: { data: GithubRepoObj[] } = await axios.get(baseURL['github'] + endPoint, {
-			headers: {
-				'Accept': 'application/vnd.github+json',
-				'Authorization': `Bearer ${access_key}`
+		const query: string = `
+			query GetUserRepositories {
+				viewer {
+					repositories(
+						first: ${perPage}
+						after: ${endCursor ? `"${endCursor}"` : null}
+						affiliations: [OWNER, ORGANIZATION_MEMBER, COLLABORATOR]
+						ownerAffiliations: [OWNER, ORGANIZATION_MEMBER, COLLABORATOR]
+					) {
+						totalCount
+						pageInfo {
+							endCursor
+							hasNextPage
+						}
+						nodes {
+							name
+							owner {
+								login
+							}
+						}
+					}
+				}
 			}
-		})
-			.catch(err => {
-				console.error(`[getUserRepositories] Error occurred while getting user repositories from GitHub API (provider-assigned id: ${authId}). Endpoint: ${endPoint}`, err.message);
-				throw err;
-			})
-		if (repos.length === 0) {
-			console.warn(`[getUserRepositories] No repositories received from GitHub API (provider-assigned id: ${authId}). Endpoint: ${endPoint}`);
+		`;
+
+		try {
+			const response = await axios.post(
+				'https://api.github.com/graphql',
+				{ query },
+				{
+					headers: {
+						Authorization: `Bearer ${access_token}`,
+					},
+				}
+			);
+			console.log(`[getUserRepositoriesForGitHub] response = ${JSON.stringify(response.data)}`)
+
+			const data = response.data.data;
+			const repos: GithubRepoObj[] = data.viewer.repositories.nodes;
+
+			if (repos.length === 0) {
+				console.warn(`[getUserRepositories] No repositories received from GitHub API (provider-assigned id: ${authId}).`);
+				break;
+			}
+
+			const allGitHubRepoIdentifiers = repos.map(repo => ({
+				repo_provider: supportedProviders[0],
+				repo_owner: repo.owner.login,
+				repo_name: repo.name,
+			}));
+
+			allGitHubRepositories.push(...allGitHubRepoIdentifiers);
+
+			hasNextPage = data.viewer.repositories.pageInfo.hasNextPage;
+			endCursor = data.viewer.repositories.pageInfo.endCursor;
+		} catch (err) {
+			console.error(`[getUserRepositories] Error occurred while getting user repositories from GitHub GraphQL API (provider-assigned id: ${authId}).`, err);
+			throw err;
+		}
+	} while (hasNextPage);
+
+	// Sort repositories by repo_owner and then by repo_name
```

Comment:
Low LOGAF

we might want to sort by activity or last updated as well. But this also looks good.